File: php/crypto/RSAEncryption.php
<?php
$text = "Very secret information";
echo "Beginning; text to encrypt: " . $text, PHP_EOL;

// Generate public and private keys
echo "Generating key pair...", PHP_EOL;
$config = array(
    "private_key_bits" => 2048,
    "private_key_type" => OPENSSL_KEYTYPE_RSA,
);
$res = openssl_pkey_new($config);
$public_key = openssl_pkey_get_details($res)["key"];
openssl_pkey_export($res, $private_key);
echo "Public key: " . $public_key, PHP_EOL;
echo "Private key: " . $private_key, PHP_EOL;

// Encrypt the message
echo "Encrypting...", PHP_EOL;
openssl_public_encrypt($text, $cipher_text, $public_key);
echo "Encrypted message: " . bin2hex($cipher_text), PHP_EOL;

// Decrypt the message
echo "Decrypting...", PHP_EOL;
openssl_private_decrypt($cipher_text, $plain_text, $private_key);
echo "Decrypted message: " . $plain_text, PHP_EOL;
?>

File: java/src/org/ggolawski/security/utils/crypto/RSAEncryption.java
package org.ggolawski.security.utils.crypto;

import java.security.GeneralSecurityException;
import java.security.KeyPair;
import java.security.PrivateKey;
import java.security.PublicKey;

import javax.crypto.Cipher;

import org.apache.commons.codec.binary.Hex;

public class RSAEncryption {
	public static void main(String[] args) throws GeneralSecurityException {
		String text = "Very secret information";
		System.out.println("Beginning; text to encrypt: " + text);
		
		// Generate public and private keys
		System.out.println("Generating key pair...");
	    KeyPair keyPair = Utils.generateKeyPair("RSA", 2048);
	    PublicKey publicKey = keyPair.getPublic();
	    PrivateKey privateKey = keyPair.getPrivate();
	    System.out.println("Public key: " + Hex.encodeHexString(publicKey.getEncoded()));
	    System.out.println("Private key: " + Hex.encodeHexString(privateKey.getEncoded()));
	    
	    Cipher cipher = Cipher.getInstance("RSA");
	    
	    // Encrypt the message
	    System.out.println("Encrypting...");
        cipher.init(Cipher.ENCRYPT_MODE, publicKey);  
        byte[] cipherText = cipher.doFinal(text.getBytes());
        System.out.println("Encrypted message: " + Hex.encodeHexString(cipherText));
        
        // Decrypt the message
        System.out.println("Decrypting...");
        cipher.init(Cipher.DECRYPT_MODE, privateKey);
        byte[] plainText = cipher.doFinal(cipherText);
        System.out.println("Decrypted message: " + new String(plainText));
	}

	
}
